src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -335,9 +335,19 @@ internal void LogBuffer(int size)
         private SocketError FinishOperationAccept(Internals.SocketAddress remoteSocketAddress)
         {
             System.Buffer.BlockCopy(_acceptBuffer!, 0, remoteSocketAddress.Buffer, 0, _acceptAddressBufferCount);
+            // We already assigned a non-null socket to this in GetOrCreateFunc
+            Socket sukru = _acceptSocket!;
             _acceptSocket = _currentSocket!.CreateAcceptSocket(
                 SocketPal.CreateSocket(_acceptedFileDescriptor),
                 _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
+            sukru.DisposeHandle();
+            sukru.CopyStateFromSource(_acceptSocket);
+            // We keep this socket to make clean-up.
+            Socket temp = _acceptSocket;
+            _acceptSocket = sukru;
+            temp.ClearHandle();
```

Comment:
since the helpers are one line, I would probably put it directly here so it is easier to see what is happening without need to scroll (personal preference)

Reply:
I'll give it a try, but as far as I remember, there were some problems with access modifiers.

Reply:
Yes, since those fields are private, I can't access them directly, so I wrote some helpers.